db/mongo_test.go

```diff
@@ -0,0 +1,302 @@
+		Workload: bson.M(map[string]interface{}{"foo": "bar"}),
+		CronTime: cron6.CronTime,
+		TimeZone: cron6.TimeZone,
+		Created:  cron6.Created,
+	}
+	assert.Equal(t, mc2.toCronJob(), cron6)
+
+}
+
+func TestToMongoCron(t *testing.T) {
+	assert.Equal(t, cron1.ID, "")
+	mc1 := cron1.toMongoCronJob()
+	var id bson.ObjectId
+	expected := mongoCronJob{
+		ID:       id,
+		IsActive: true,
+		Function: "test1",
+		Workload: "",
+		CronTime: "* * * * * *",
+		TimeZone: trialTimeZone,
+		Created:  trialCreateTime,
+	}
+	assert.Equal(t, mc1, expected)
+
+	assert.NotEqual(t, cron5.ID, "")
+	mc2 := cron5.toMongoCronJob()
+	expected = mongoCronJob{
+		ID:       bson.ObjectId(mc2.ID),
+		IsActive: true,
+		Function: "test4",
+		Workload: "",
+		CronTime: "6 5 4 3 2 1",
+		TimeZone: trialTimeZone,
+		Created:  trialCreateTime,
+	}
+	assert.Equal(t, mc2, expected)
+}
+
+func TestAddJob(t *testing.T) {
+	dbName := "testAddJob"
+	testURL := mongoTestURL(dbName)
+	database, dialErr := NewMongoDB(testURL, dbName)
+	assert.NoError(t, dialErr)
+	defer database.session.Close()
+	defer dropDatabase(testURL)
+
+	addErr := database.AddJob(cron1)
+	assert.NoError(t, addErr)
+
+	var mc mongoCronJob
+	cronCollection := database.getCronCollection(nil)
+	queryErr := cronCollection.Find(bson.M{"function": cron1.Function}).One(&mc)
+	assert.NoError(t, queryErr)
+
+	cron1.ID = mc.ID.Hex()
+	cron1.Created = mc.Created
+	assert.Equal(t, cron1, mc.toCronJob())
+	cron1.ID = ""
+}
+
+func TestDeleteJob(t *testing.T) {
+	dbName := "testDeleteJob"
+	testURL := mongoTestURL(dbName)
+	database, dialErr := NewMongoDB(testURL, dbName)
+	assert.NoError(t, dialErr)
+	defer database.session.Close()
+	defer dropDatabase(testURL)
+
+	addErr := database.AddJob(cron1)
+	assert.NoError(t, addErr)
+
+	var mc mongoCronJob
+	cronCollection := database.getCronCollection(nil)
+	queryErr := cronCollection.Find(bson.M{"function": cron1.Function}).One(&mc)
+	assert.NoError(t, queryErr)
+
+	deleteErr := database.DeleteJob(mc.ID.Hex())
+	assert.NoError(t, deleteErr)
+
+	count, countErr := cronCollection.Count()
+	assert.NoError(t, countErr)
+	assert.Equal(t, count, 0)
+}
+
+func TestUpdateJob(t *testing.T) {
+	dbName := "testUpdateJob"
+	testURL := mongoTestURL(dbName)
+	database, dialErr := NewMongoDB(testURL, dbName)
+	assert.NoError(t, dialErr)
+	defer database.session.Close()
+	defer dropDatabase(testURL)
+
+	addErr := database.AddJob(cron1)
+	assert.NoError(t, addErr)
+
+	var mcBefore mongoCronJob
+	cronCollection := database.getCronCollection(nil)
+	queryErr := cronCollection.Find(bson.M{"function": cron1.Function}).One(&mcBefore)
+	assert.NoError(t, queryErr)
+
+	modJob := cron2
+	modJob.ID = mcBefore.ID.Hex()
+
+	updateErr := database.UpdateJob(modJob)
+	assert.NoError(t, updateErr)
+
+	var mcAfter mongoCronJob
+	queryErr = cronCollection.FindId(bson.ObjectIdHex(modJob.ID)).One(&mcAfter)
+	assert.NoError(t, queryErr)
+	modJob.Created = mcAfter.Created
+	assert.Equal(t, mcAfter.toCronJob(), modJob)
```

Comment:
in fact I'd say all of the tests in this file should construct the DB interface and then only use that

basically you want to take the interface for a spin and see if it has sensible behavior. You should be able to do this without direct calls to mongo